xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -758,8 +758,13 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
       continue;
 
     m_codecname = codec_info.getName();
-    if (!CServiceBroker::GetDecoderFilterManager()->isValid(m_codecname, m_hints))
+    if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+            CSettings::SETTING_VIDEOPLAYER_USEDECODERFILTER) &&
+        (!CServiceBroker::GetDecoderFilterManager()->isValid(m_codecname, m_hints)))
+    {
+      CLog::Log(LOGDEBUG, "CDVDVideoCodecAndroidMediaCodec::Open Codec disabled: {}", m_codecname);
```

Comment:
Nitpicking: I would add a ':' after 'Open' for better readability. I had to read it two times to actually understand the content.

Reply:
I followed the style of the rest of the logs of the Open() function: "CDVDVideoCodecAndroidMediaCodec::Open + text".

I know `CLog::LogF`, the problem is that this function in Android has a different behaviour and only shows the function name without the class name.

Reply:
why not change the behaviour of LogF define based on android platform so to have LogF prints consistent between all systems?
in case not to do on this pr

Reply:
It would be nice, I was reading some time ago about that and I remember that it is not easy to get the class name at runtime in Android, if someone knows more about this it would be interesting.

Reply:
i tried investigate but seem there is no builtin way
there is __PRETTY_FUNCTION__ but provide also return value and arguments, too much info
an idea is to remove info from __PRETTY_FUNCTION__ string and so make a custom method to do it
would however cause a slight overload which should be cut off when debugging is disabled
no easy way